### PR TITLE
Fix linting

### PR DESCRIPTION
**Pull request by wipfli**
This pull request fixes the remaining 30 linter errors. I removed `src/index.d.ts` since we won't need it anymore.

 - [🐳  ] confirm your changes do not include backports from Mapbox projects (unless with compliant license)
 - [ ] briefly describe the changes in this PR
 - [ ] include before/after visuals or gifs if this PR includes visual changes
 - [ ] write tests for all new functionality
 - [ ] document any changes to public APIs
 - [ ] post benchmark scores
 - [ ] manually test the debug page
 - [ ] apply changelog label ('bug', 'feature', 'docs', etc) or use the label 'skip changelog'
 - [ ] add an entry inside this element for inclusion in the `maplibre-gl-js` changelog: `<changelog></changelog>`

### CODE REVIEW

File: src/style-spec/format.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 
-import reference from './reference/latest.js';
+import reference from './reference/latest.ts';
```
Comment (HarelM) [inline src/style-spec/format.ts:2]: Can we remove the file extension?

File: src/util/web_worker_transfer.ts [hunk 2]
```diff
@@ -185,7 +185,7 @@ export function serialize(input: unknown, transferables?: Array<Transferable> |
         if (!klass.serialize) {
             for (const key in input) {
                 // any cast due to https://github.com/facebook/flow/issues/5393
-                if (!(input as any).hasOwnProperty(key)) continue;
+                if (!(input as any).hasOwnProperty(key)) continue; // eslint-disable-line no-prototype-builtins
```
Comment (HarelM) [inline src/util/web_worker_transfer.ts:188]: Why did you choose to disable this and not fix it like you did with the others?

**Comment by HarelM** [comment]
Nice work man! :-)

**Review by HarelM — APPROVED** [review]
Besides the comments I had this is approved...

File: src/util/web_worker_transfer.ts (hunk 2 above)
Comment (wipfli) [reply src/util/web_worker_transfer.ts:188]: Yes, 234c200.

File: src/style-spec/format.ts (hunk 1 above)
Comment (wipfli) [reply src/style-spec/format.ts]: Yes, see 234c200.

File: src/util/web_worker_transfer.ts (hunk 2 above)
Comment (wipfli) [reply src/util/web_worker_transfer.ts:188]: Because I don't understand how this line works.

**Comment by wipfli** [comment]
I did a rebase merge because I don't  want to squash my commits such that I can later have the possibility to regroup changes.

**Pull request merged**